spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Container.scala

```diff
@@ -29,33 +29,21 @@ import scala.reflect.ClassTag
  * [[Container]] is an abstract [[AbstractModule]] class which
  * declares methods defined in all containers. A container usually
  * contain some other modules in the `modules` variable. It overrides
- * many module methods such that calls are propogated to the contained
+ * many module methods such that calls are propagated to the contained
  * modules.
  *
  * @tparam A Input data type
  * @tparam B Output data type
  * @tparam T Numeric type. Only support float/double now
  */
 @SerialVersionUID(- 2120105647780417237L)
-abstract class Container[A <: Activity : ClassTag,
-    B <: Activity : ClassTag, T: ClassTag](
+abstract class Container[A <: Activity : ClassTag, B <: Activity : ClassTag, T: ClassTag](
   implicit ev: TensorNumeric[T]) extends AbstractModule[A, B, T] {
 
   // list of sub modules
   val modules: ArrayBuffer[AbstractModule[Activity, Activity, T]]
```

Comment:
user can still add module if we expose `ArrayBuffer` here or we can put it into constructor and make it immutable.

Reply:
It is not an immutable object... I think maybe we can make a defense copy here. The user can get a copy of modules but cannot change the original one.

However, this still breaks the original API behavior. Maybe break some client code.